source/contentscript.js

```diff
@@ -1,11 +1,14 @@
-// when asked tell the extension about the SPDY/HTTP2 status of current page
-
+// determineConnectionInfo returns the protocol used to connect to the first hop
+// for the first navigation event available.
 function determineConnectionInfo() {
-  var loadTimes = chrome.loadTimes();
-  return {
-    spdy: loadTimes.wasFetchedViaSpdy,
-    type: loadTimes.npnNegotiatedProtocol || loadTimes.connectionInfo
-  };
+  if (performance && performance.getEntriesByType) {
+    let entry = performance.getEntriesByType('navigation')[0];
+    let proto = entry && entry.nextHopProtocol;
```

Comment:
Looking at the [MDN browser compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/nextHopProtocol#Browser_compatibility) shows that only Firefox currently supports `nextHopProtocol`, so we should probably keep both methods for now.

Reply:
It doesn't support the previous API either, so there's no loss in removing it. It will start working in Firefox 58.

Reply:
Sorry, I misread. The MDN compatibility table is outdated. It's been [supported in Chrome since M61](https://www.chromestatus.com/features/5650461096935424).

Reply:
I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1440515 because Fx58 has been released and this is broken.

Reply:
It works on my machine:
![`PerformanceNavigationTiming` works for me on my machine](https://user-images.githubusercontent.com/3889017/36622778-b458a492-18ff-11e8-8a3f-eac7feb9f7ae.png)